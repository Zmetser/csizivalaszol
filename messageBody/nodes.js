/**
 * @flow
 *
 * All the usable nodes
 */

export type InlineStyle = 'Italic' | 'Bold' | 'Struck' | 'Small' | 'Code' | 'Subscript' | 'Superscript' | 'Underline' | 'Normal'

const TextNode = (value: string, styles: Array<InlineStyle>) => ({
  type: 'Text',
  value,
  styles
})

const LineBreak = () => ({ type: 'LineBreak' })

const Link = (value: Array<TextNode>, href: string, target?: ?string) => ({
  type: 'Link',
  value,
  href,
  target
})

export type InlineNode = TextNode | LineBreak | Link;

type ImageAttributes = { src: string, alt: string, width: number, height: number }
const Image = ({ src, alt, width, height }: ImageAttributes) => ({
  type: 'Image',
  src,
  alt,
  width,
  height
})

const Paragraph = (value: Array<InlineNode>) => ({
  type: 'Paragraph',
  value
})

module.exports = {
  TextNode,
  Paragraph,
  LineBreak,
  Link,
  Image
}
