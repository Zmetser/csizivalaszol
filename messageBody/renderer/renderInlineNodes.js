/**
 * @flow
 */

import * as React from 'react'
import { UnknownNodeError } from '../errors'

import type { InlineNode } from '../types'

import {
  Text,
  Link
} from '../components'

function createInlineNode (node: InlineNode, index): React.Element<*> {
  switch (node.type) {
    case 'Text': return Text(node, index)
    case 'Link': return <Link {...node} key={index} />
    default: throw new UnknownNodeError(node)
  }
}
export default function renderInlineNodes (nodes: Array<InlineNode>): Array<React.Element<*>> {
  return nodes.map(createInlineNode)
}
