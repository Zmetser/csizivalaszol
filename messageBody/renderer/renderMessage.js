/**
 * @flow
 */

import * as React from 'react'
import {
  Paragraph,
  UnorderedList,
  OrderedList
} from '../components'

import type { BlockNode } from '../types.js'
import { UnknownNodeError } from '../errors'

export default function render (message: Array<BlockNode>): Array<React.Element<'p' | 'ul' | 'ol'>> {
  return message.map((node, index): React.Element<*> => {
    switch (node.type) {
      case 'Paragraph': return Paragraph({ value: node.value }, index)
      case 'List':
        return (node.listType === 'Bullet')
          ? UnorderedList({ items: node.items }, index)
          : OrderedList({ items: node.items }, index)
      default: throw new UnknownNodeError(node)
    }
  })
}
