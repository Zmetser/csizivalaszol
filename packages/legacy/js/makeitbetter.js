function del() {
  $elem = document.getElementById('page');
  $elem.value = "";
}