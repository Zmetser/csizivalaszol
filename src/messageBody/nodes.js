/**
 * All the usable nodes
 */

import type {
  ImageAttributes,
  ImageNode,
  InlineNode,
  InlineStyle,
  LineBreakNode,
  LinkNode,
  ParagraphNode,
  TextNode,
  ListType,
  ListItem,
  ListNode
} from './types'

export const createTextNode = (value: string, styles: Array<InlineStyle>): TextNode => ({
  type: 'Text',
  value,
  styles
})

export const createLineBreak = (): LineBreakNode => ({ type: 'LineBreak' })

export const createLink = (value: Array<TextNode>, href: string, target?: ?string): LinkNode => ({
  type: 'Link',
  value,
  href,
  target
})

export const createImage = ({ src, alt, width, height }: ImageAttributes): ImageNode => ({
  type: 'Image',
  src,
  alt,
  width,
  height
})

export const createParagraph = (value: Array<InlineNode>): ParagraphNode => ({
  type: 'Paragraph',
  value
})

export function createList<T: ListType> (items: Array<ListItem>, listType: T): ListNode<T> {
  return {
    type: 'List',
    listType,
    items
  }
}
