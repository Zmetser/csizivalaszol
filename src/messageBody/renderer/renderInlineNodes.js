/**
 * @flow
 */

import * as React from 'react'
import { UnknownNodeError } from '../errors'

import type { InlineNode } from '../types'

import {
  Text,
  Link,
  LineBreak
} from '../components'

function createInlineNode (node: InlineNode, index): React.Element<*> {
  switch (node.type) {
    case 'Text': return <Text node={node} key={index} />
    case 'Link': return <Link node={node} key={index} />
    case 'LineBreak': return <LineBreak key={index} />
    default: throw new UnknownNodeError(node)
  }
}
export default function renderInlineNodes (nodes: Array<InlineNode>): Array<React.Element<*>> {
  return nodes.map(createInlineNode)
}
