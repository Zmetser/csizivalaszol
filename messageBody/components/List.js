/**
 * @flow
 */

import * as React from 'react'
import type { ListNode } from '../types'

import renderInlineNodes from '../renderer/renderInlineNodes'

const renderListItem = (value, index) => (
  <li key={index}>{renderInlineNodes(value)}</li>
)

export function UnorderedList ({ items }: $Shape<ListNode<'Bullet'>>, key: React.Key): React.Element<'ul'> {
  return <ul key={key}>{items.map((value, index) => renderListItem(value, index))}</ul>
}

export function OrderedList ({ items }: $Shape<ListNode<'Number'>>, key: React.Key): React.Element<'ol'> {
  return <ol key={key}>{items.map((value, index) => renderListItem(value, index))}</ol>
}
