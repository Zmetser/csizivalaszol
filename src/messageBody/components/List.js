/**
 * @flow
 */

import * as React from 'react'
import type { ListNode, ListTypes } from '../types'

import { UnknownNodeError } from '../errors'

import renderInlineNodes from '../renderer/renderInlineNodes'

const renderListItem = (value, index) => (
  <li key={index}>{renderInlineNodes(value)}</li>
)

export function UnorderedList ({ node }: { node: ListNode<'Bullet'> }): React.Element<'ul'> {
  const { items } = node
  return <ul>{items.map(renderListItem)}</ul>
}

export function OrderedList ({ node }: { node: ListNode<'Number'> }): React.Element<'ol'> {
  const { items } = node
  return <ol>{items.map(renderListItem)}</ol>
}

export default function List ({ node }: { node: ListTypes }): React.Element<typeof UnorderedList> | React.Element<typeof OrderedList> {
  switch (node.listType) {
    case 'Number': return <OrderedList node={node} />
    case 'Bullet': return <UnorderedList node={node} />
    default: {
      throw new UnknownNodeError(node)
    }
  }
}
