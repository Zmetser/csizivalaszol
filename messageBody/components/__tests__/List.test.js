/**
 * @flow
 */
import renderer from 'react-test-renderer'

import { createTextNode } from '../../nodes.js'

import { OrderedList, UnorderedList } from '../List'

const fooText = createTextNode('Foo', [])

describe('<OrderedList />', () => {
  test('with a single item', () => {
    const element = renderer.create(
      OrderedList({ items: [[fooText]] }, 0)
    )
    expect(element.toJSON()).toMatchSnapshot()
  })
  test('with 2 items', () => {
    const element = renderer.create(
      OrderedList({ items: [[fooText], [fooText]] }, 0)
    )
    expect(element.toJSON()).toMatchSnapshot()
  })
})

describe('<UnorderedList />', () => {
  test('with a single item', () => {
    const element = renderer.create(
      UnorderedList({ items: [[fooText]] }, 0)
    )
    expect(element.toJSON()).toMatchSnapshot()
  })
  test('with 2 items', () => {
    const element = renderer.create(
      UnorderedList({ items: [[fooText], [fooText]] }, 0)
    )
    expect(element.toJSON()).toMatchSnapshot()
  })
})
