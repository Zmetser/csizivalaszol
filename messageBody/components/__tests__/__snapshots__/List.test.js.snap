// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderedList /> with 2 items 1`] = `
<ol>
  <li>
    Foo
  </li>
  <li>
    Foo
  </li>
</ol>
`;

exports[`<OrderedList /> with a single item 1`] = `
<ol>
  <li>
    Foo
  </li>
</ol>
`;

exports[`<UnorderedList /> with 2 items 1`] = `
<ul>
  <li>
    Foo
  </li>
  <li>
    Foo
  </li>
</ul>
`;

exports[`<UnorderedList /> with a single item 1`] = `
<ul>
  <li>
    Foo
  </li>
</ul>
`;
